File: wallet/src/main/java/chatox/wallet/api/response/UploadResponse.java
package chatox.wallet.api.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UploadResponse<T> {
    private String id;
    private String name;
    private String extension;
    private String mimeType;
    private T meta;
    private UploadType type;
    private String uri;
    private String originalName;
}

File: user/src/main/kotlin/chatox/user/api/response/UploadResponse.kt
package chatox.user.api.response

import chatox.user.domain.ImageUploadMetadata
import chatox.user.domain.UploadType

data class UploadResponse<MetadataType>(
        val id: String,
        val name: String,
        val extension: String,
        val mimeType: String,
        val meta: MetadataType,
        val type: UploadType,
        val uri: String,
        val originalName: String
)
